internal/databases/postgres/daemon_handler.go

```diff
@@ -0,0 +1,176 @@
+package postgres
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path"
+
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/utility"
+)
+
+type SocketMessageType byte
+
+const (
+	CheckType    SocketMessageType = 'C'
+	FileNameType SocketMessageType = 'F'
+)
+
+type SocketMessageHandler interface {
+	Handle(messageBody []byte) error
+}
+
+type ArchiveMessageHandler struct {
+	messageType SocketMessageType
+	fd          net.Conn
+	uploader    *WalUploader
+}
+
+func (h *CheckMessageHandler) Handle(messageBody []byte) error {
+	_, err := h.fd.Write([]byte{'O'})
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Error on writing in socket: %v \n", err)
+		return err
+	}
+	tracelog.InfoLogger.Println("Successful configuration check")
+	return nil
+}
+
+type CheckMessageHandler struct {
+	messageType SocketMessageType
+	fd          net.Conn
+	uploader    *WalUploader
+}
+
+func (h *ArchiveMessageHandler) Handle(messageBody []byte) error {
+	err := messageValidation(messageBody)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Incorrect message: %v\n", err)
```

Comment:
You can just log the error in the upper level.
Here you can use the fmt.Errorf("msg: %w", err) to wrap error with some message.

For example:
```go
if err != nil {
    return fmt.Errorf("message validation failed: %w", err)
}
```

Reply:
Same issue in other places below.